jax/experimental/jax2tf/impl_no_xla.py

```diff
@@ -217,19 +232,27 @@ def _conv_general_dilated(
     preferred_element_type: Optional[DType],
     _in_avals: Sequence[core.ShapedArray], _out_aval: core.ShapedArray):
   """Implementation of lax.conv_general_dilated_p using XlaConv."""
+  # In presence of shape polymorphism, lhs.shape and rhs.shape may contain
+  # None. The actual dimension polynomial shapes are in _in_avals.
   del lhs_shape, rhs_shape, precision  # Unused arguments.
-  out_shape = jax2tf._aval_to_tf_shape(_out_aval)
-  _validate_spatial_dimensions(len(lhs.shape) - 2)
-  is_conv1d = len(lhs.shape) - 2 == 1
+  lhs_shape, rhs_shape = _in_avals[0].shape, _in_avals[1].shape
+  jax2tf._assert_matching_abstract_shape(lhs, lhs_shape)
+  jax2tf._assert_matching_abstract_shape(rhs, rhs_shape)
```

Comment:
Consider updating _validate_spatial_dimensions to consume (lhs, lhs_shape, rhs, rhs_shape) and moving these asserts into it; they certainly with fit the function name, and it'll simplify reading the main function a little bit.